### DISCUSSION TITLE
Need Help with Creating Custom Chain Buttons for Wallet Connection

### DISCUSSION THREAD

**Discussion by fahimfaisaal** [WalletConnect/walletconnect-monorepo#3771] [Developer Support] [answered]
Hello 👋🏼 ,

I need some help with integrating two custom buttons in my application. These buttons will allow users to connect their Ethereum and Polygon wallets seamlessly. I'm using the `useWeb3Modal` hook to open a modal for wallet connections, but I'm encountering a few challenges.

Here's the problem I'm facing:

1. **Default Chain Issue:** By default, the `useWeb3Modal` hook sets the first chain from the chains array or the `defaultChain` property. This behavior doesn't suit my use case, as I want users to be able to select a dedicated network only after clicking the corresponding button.

2. **Undefined Network:** I've tried to use the `useSwitchNetwork` hook to switch to a dedicated network, but it's showing me "undefined" before the user is connected to a wallet. This creates a usability issue as users should be able to choose their network at the time of connecting the wallet.

## What I Want:

I'm looking for a solution that allows users to connect their Ethereum and Polygon wallets by simply clicking the respective buttons with the dedicated chain name, rather than having to go through a separate network selection step. i.e. `open({ view: 'Networks' })`.

I'm open to any suggestions, code snippets, or guidance on how to achieve this user-friendly wallet connection process. 

Thank you in advance for your help! 🚀

### Versions
```json
{
   "@web3modal/wagmi": "^3.0.2",
    "viem": "^1.15.4",
    "wagmi": "^1.4.3"
}

**Comment by glitch-txs** [comment, marked as answer]
Hi, this is not possible at the moment, however you can open an issue in Web3Modal repository with this feature request

**Reply by fahimfaisaal** [reply]
Thanks for the valuable response 👍🏼.
